### PR TITLE
[JSDoc] Add missing documentation for botbuilder-dialogs-adaptive Actions folder 2/3

### PR DESCRIPTION
**Pull request by iscaltritti-sw**
Addresses # 2602

## Description
This PR adds all the missing JSDoc documentation for [botbuilder-dialogs-adaptive](https://github.com/microsoft/botbuilder-js/tree/main/libraries/botbuilder-dialogs-adaptive) based on the errors thrown. Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet).

## Specific Changes

- Ported documentation from botbuilder-dotnet
- Added `@private` for private methods.
- Added JSDoc comments in **13 files**
- Fixes _missing semicolon_ warnings.

## Testing
In the following image there is a sneak peek about the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

**Review by sw-joelmut — APPROVED** [review]

**Review by santgr11 — APPROVED** [review]

File: libraries/botbuilder-dialogs-adaptive/src/actions/editArray.ts [hunk 1]
```diff
@@ -63,6 +82,13 @@ export class EditArray<O extends object = {}> extends Dialog<O> {
      */
     public disabled?: BoolExpression;
 
+    /**
+     * Starts a new dialog and pushes it onto the dialog stack.
+     * @param dc The `DialogContext` for the current turn of conversation.
+     * @param result Optional, value returned from the dialog that was called. The type
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/actions/editArray.ts]: Same here

File: libraries/botbuilder-dialogs-adaptive/src/actions/emitEvent.ts [hunk 2]
```diff
@@ -38,6 +55,13 @@ export class EmitEvent<O extends object = {}> extends Dialog<O> {
      */
     public disabled?: BoolExpression;
 
+    /**
+     * Starts a new dialog and pushes it onto the dialog stack.
+     * @param dc The `DialogContext` for the current turn of conversation.
+     * @param result Optional, value returned from the dialog that was called. The type
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/actions/emitEvent.ts]: same.

File: libraries/botbuilder-dialogs-adaptive/src/actions/endDialog.ts [hunk 3]
```diff
@@ -30,6 +34,13 @@ export class EndDialog<O extends object = {}> extends Dialog<O> {
      */
     public disabled?: BoolExpression;
 
+    /**
+     * Starts a new dialog and pushes it onto the dialog stack.
+     * @param dc The `DialogContext` for the current turn of conversation.
+     * @param result Optional, value returned from the dialog that was called. The type
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/actions/endDialog.ts]: Same.

File: libraries/botbuilder-dialogs-adaptive/src/actions/endTurn.ts [hunk 4]
```diff
@@ -9,19 +9,35 @@ import { DialogTurnResult, Dialog, DialogContext } from 'botbuilder-dialogs';
...
 export class EndTurn<O extends object = {}> extends Dialog<O> {
     /**
      * An optional expression which if is true will disable this action.
      */
     public disabled?: BoolExpression;
 
+    /**
+     * Starts a new dialog and pushes it onto the dialog stack.
+     * @param dc The `DialogContext` for the current turn of conversation.
+     * @param result Optional, value returned from the dialog that was called. The type
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/actions/endTurn.ts]: Same in here.

File: libraries/botbuilder-dialogs-adaptive/src/actions/forEachPage.ts [hunk 5]
```diff
@@ -54,10 +60,21 @@ export class ForEachPage<O extends object = {}> extends ActionScope<O> {
...
+     * @returns The child dialog dependencies.
+     */
     public getDependencies(): Dialog[] {
         return this.actions;
     }
 
+    /**
+     * Starts a new dialog and pushes it onto the dialog stack.
+     * @param dc The `DialogContext` for the current turn of conversation.
+     * @param result Optional. Value returned from the dialog that was called. The type
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/actions/forEachPage.ts:74]: Same in here.

File: libraries/botbuilder-dialogs-adaptive/src/actions/getActivityMembers.ts [hunk 6]
```diff
@@ -8,8 +8,17 @@
...
+/**
+ * Calls `BotFrameworkAdapter.getActivityMembers()` and sets the result to a memory property.
+ */
 export class GetActivityMembers<O extends object = {}> extends Dialog {
     public constructor();
+
+    /**
+     * Initializes a new instance of the `GetActivityMembers` class.
+     * @param activityId The expression to get the value to put into property path.
+     * @param property Property path to put the value in.
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/actions/getActivityMembers.ts:20]:
```suggestion
     * @param activityId Optional. The expression to get the value to put into property path.
     * @param property Optional. Property path to put the value in.
```

File: libraries/botbuilder-dialogs-adaptive/src/actions/getActivityMembers.ts [hunk 7]
```diff
@@ -31,6 +40,13 @@ export class GetActivityMembers<O extends object = {}> extends Dialog {
      */
     public disabled?: BoolExpression;
 
+    /**
+     * Starts a new dialog and pushes it onto the dialog stack.
+     * @param dc The `DialogContext` for the current turn of conversation.
+     * @param result Optional. Value returned from the dialog that was called. The type
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/actions/getActivityMembers.ts:46]: same in here.

File: libraries/botbuilder-dialogs-adaptive/src/actions/getConversationMembers.ts [hunk 8]
```diff
@@ -25,6 +33,13 @@ export class GetConversationMembers<O extends object = {}> extends Dialog<O> {
      */
     public disabled?: BoolExpression;
 
+    /**
+     * Starts a new dialog and pushes it onto the dialog stack.
+     * @param dc The `DialogContext` for the current turn of conversation.
+     * @param result Optional. Value returned from the dialog that was called. The type
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/actions/getConversationMembers.ts:39]: Same.

File: libraries/botbuilder-dialogs-adaptive/src/actions/gotoAction.ts [hunk 9]
```diff
@@ -26,6 +34,13 @@ export class GotoAction<O extends object = {}> extends Dialog<O> {
      */
     public disabled?: BoolExpression;
 
+    /**
+     * Starts a new dialog and pushes it onto the dialog stack.
+     * @param dc The `DialogContext` for the current turn of conversation.
+     * @param result Optional. Value returned from the dialog that was called. The type
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/actions/gotoAction.ts:40]: same.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Comment by iscaltritti-sw** [comment]
Solved, @ceciliaavila. Thanks!

**Review by ceciliaavila — APPROVED** [review]

File: libraries/botbuilder-dialogs-adaptive/src/actions/deleteProperties.ts [hunk 10]
```diff
@@ -37,7 +37,7 @@ export class DeleteProperties<O extends object = {}> extends Dialog<O> {
 
     /**
      * Called when the dialog is started and pushed onto the dialog stack.
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/actions/deleteProperties.ts:39]: Same changes for the other two Actions PRs.
We should add the dialog reference in multiple files

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

File: libraries/botbuilder-dialogs-adaptive/src/actions/deleteProperties.ts (hunk 10 above)
Comment (iscaltritti-sw) [reply libraries/botbuilder-dialogs-adaptive/src/actions/deleteProperties.ts]: Solved!

**Review by sw-joelmut — APPROVED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
@iscaltritti-sw Please solve the conflicts and verify that the build passes so we can promote it

**Comment by iscaltritti-sw** [comment]
Conflicts solved, build passes. All ready to go, @denscollo

**Comment by denscollo** [comment]
promoted

**Pull request closed without merging**